source/command-monitoring/tests/unified/redacted-commands.yml

```diff
@@ -177,76 +210,131 @@ tests:
               command: { copydb: { $$exists: false } }
 
   - description: "hello with speculative authenticate"
+    runOnRequirements:
+      - minServerVersion: "4.9"
     operations:
       - name: runCommand
         object: *database
         arguments:
           commandName: hello
           command:
-            hello: "private"
-            speculativeAuthenticate: "foo"
-        expectError:
-          isError: true
+            hello: 1
+            speculativeAuthenticate:
+              saslStart: 1
+    expectEvents:
+      - client: *client
+        events:
+          - commandStartedEvent:
+              commandName: hello
+              command:
+                hello: { $$exists: false }
+                speculativeAuthenticate: { $$exists: false }
+          - commandSucceededEvent:
+              commandName: hello
+              reply:
+                # Even though authentication above fails and the reply does not
+                # contain sensitive information, we're expecting the reply to be
+                # redacted as well.
+                isWritablePrimary: { $$exists: false }
+                # This assertion will currently always hold true since we're
+                # not expecting successful authentication, in which case this
+                # field is missing anyways.
+                speculativeAuthenticate: { $$exists: false }
+
+  - description: "legacy hello with speculative authenticate"
+    operations:
```

Comment:
Would it make sense to give this test a `maxServerVersion` requirement, or are there no plans for the server to remove support for `isMaster` entirely in the foreseeable future?

Reply:
I know of no plans to drop legacy hello, and even the work to deprecated legacy opcodes makes an exception for using `OP_QUERY` to send an initial handshake command, so I don't think we'll have an issue.

If the legacy hello command were indeed removed from the server, I think some more of our tests need updating to account for this so I'd rather defer for the time being.